app/Livewire/Project/Service/Domains.php

```
@@ -0,0 +1,1289 @@
+            $currentUrls = collect($this->domainRows)
+                ->where('service_application_id', $app->id)
+                ->pluck('url')
+                ->all();
+            $statuses = array_intersect_key($statuses, array_flip($currentUrls));
+
+            $app->domain_dns_statuses = $statuses === [] ? null : $statuses;
+            $app->save();
+        }
+
+        $this->service->load('applications');
+    }
```

PHPMD (2.15.0)

[warning] The method persistAllDomainDnsStatuses() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)